test/core/const.wast

```diff
@@ -135,3 +141,300 @@
+  (module quote "(func (f32.const nan:1) drop)")
+  "unknown operator"
+)
+(assert_malformed
+  (module quote "(func (f64.const nan:1) drop)")
+  "unknown operator"
+)
+
+(assert_malformed
+  (module quote "(func (f32.const nan:0x0) drop)")
+  "constant out of range"
+)
+(assert_malformed
+  (module quote "(func (f64.const nan:0x0) drop)")
+  "constant out of range"
+)
+
+(assert_malformed
+  (module quote "(func (f32.const nan:0x80_0000) drop)")
+  "constant out of range"
+)
+(assert_malformed
+  (module quote "(func (f64.const nan:0x10_0000_0000_0000) drop)")
+  "constant out of range"
+)
+
+
+;; Signs on special values
+
+(module
+  (func (export "f32") (param f32) (result i32)
+    (i32.reinterpret_f32 (local.get 0))
+  )
+  (func (export "f64") (param f64) (result i64)
+    (i64.reinterpret_f64 (local.get 0))
+  )
+)
+
+(assert_return (invoke "f32" (f32.const 0)) (i32.const 0x0000_0000))
+(assert_return (invoke "f32" (f32.const +0)) (i32.const 0x0000_0000))
+(assert_return (invoke "f32" (f32.const -0)) (i32.const 0x8000_0000))
+
+(assert_return (invoke "f64" (f64.const 0)) (i64.const 0x0000_0000_0000_0000))
+(assert_return (invoke "f64" (f64.const +0)) (i64.const 0x0000_0000_0000_0000))
+(assert_return (invoke "f64" (f64.const -0)) (i64.const 0x8000_0000_0000_0000))
+
+(assert_return (invoke "f32" (f32.const inf)) (i32.const 0x7f80_0000))
+(assert_return (invoke "f32" (f32.const +inf)) (i32.const 0x7f80_0000))
+(assert_return (invoke "f32" (f32.const -inf)) (i32.const 0xff80_0000))
+
+(assert_return (invoke "f64" (f64.const inf)) (i64.const 0x7ff0_0000_0000_0000))
+(assert_return (invoke "f64" (f64.const +inf)) (i64.const 0x7ff0_0000_0000_0000))
+(assert_return (invoke "f64" (f64.const -inf)) (i64.const 0xfff0_0000_0000_0000))
+
+(assert_return (invoke "f32" (f32.const nan)) (i32.const 0x7fc0_0000))
+(assert_return (invoke "f32" (f32.const +nan)) (i32.const 0x7fc0_0000))
+(assert_return (invoke "f32" (f32.const -nan)) (i32.const 0xffc0_0000))
+
+(assert_return (invoke "f64" (f64.const nan)) (i64.const 0x7ff8_0000_0000_0000))
+(assert_return (invoke "f64" (f64.const +nan)) (i64.const 0x7ff8_0000_0000_0000))
+(assert_return (invoke "f64" (f64.const -nan)) (i64.const 0xfff8_0000_0000_0000))
+
+(assert_return (invoke "f32" (f32.const nan:0x12_3456)) (i32.const 0x7f92_3456))
+(assert_return (invoke "f32" (f32.const +nan:0x12_3456)) (i32.const 0x7f92_3456))
+(assert_return (invoke "f32" (f32.const -nan:0x12_3456)) (i32.const 0xff92_3456))
+
+(assert_return (invoke "f64" (f64.const nan:0x1_2345_6789_abcd)) (i64.const 0x7ff1_2345_6789_abcd))
+(assert_return (invoke "f64" (f64.const +nan:0x1_2345_6789_abcd)) (i64.const 0x7ff1_2345_6789_abcd))
+(assert_return (invoke "f64" (f64.const -nan:0x1_2345_6789_abcd)) (i64.const 0xfff1_2345_6789_abcd))
+
+
+;; Rounding behaviour
+
+(module
+  (func (export "f32") (param f32) (result f32) (local.get 0))
+  (func (export "f64") (param f64) (result f64) (local.get 0))
+)
+
+;; f32, small exponent
```

Comment:
We should probably have some decimal test cases here too.

Reply:
Done.